src/sentry/static/sentry/app/views/settings/account/accountIdentities.jsx

```diff
@@ -0,0 +1,89 @@
+import {Box} from 'grid-emotion';
+import React from 'react';
+
+import {disconnectIdentity} from '../../../actionCreators/account';
+import {t} from '../../../locale';
+import AsyncView from '../../asyncView';
+import Button from '../../../components/buttons/button';
+import EmptyMessage from '../components/emptyMessage';
+import Panel from '../components/panel';
+import PanelBody from '../components/panelBody';
+import PanelHeader from '../components/panelHeader';
+import Row from '../components/row';
+import SettingsPageHeader from '../components/settingsPageHeader';
+
+const ENDPOINT = '/account/identities/';
+class AccountIdentities extends AsyncView {
+  getEndpoints() {
+    return [['identities', ENDPOINT]];
+  }
+
+  getDefaultState() {
+    return {
+      identities: [],
+    };
+  }
+
+  handleDisconnect = (identity, index, e) => {
+    let {identities} = this.state;
+
+    this.setState(
+      state => {
+        let newIdentities = state.identities.slice();
+        newIdentities.splice(index, 1);
```

Comment:
what do you think about filtering it out by name or id instead of by index? seems less fragile in the face of async execution. maybe too paranoid.

Reply:
also if you keep it as index, you couuuuuuulld do `newIdentities = state.identities.filter((_,i)=> i !== index)`
Maybe i'm just biased against splice